### ISSUE TITLE
"Package does not have a signature matching the target" error when using signingConfigs

### ISSUE THREAD

**Issue by evaristokbza** [Banno/Gordon#44]
Since 1.4.0 version, Gordon task fails with the following message:

> INSTRUMENTATION_STATUS: Error=Permission Denial: starting instrumentation ComponentInfo{com.example.gordontest.test/androidx.test.runner.AndroidJUnitRunner} from pid=10898, uid=10898 not allowed because package com.example.gordontest.test does not have a signature matching the target com.example.gordontest 

After doing some tests, this stopped happening as soon as I commented out signingConfigs in gradle. 

I created an example project with just one test and Espresso dependency and the same error happens when signingConfigs is added in gradle.

**Comment by joshschriever** [comment]
Can you show an example of how you're adding signingConfigs?

**Comment by evaristokbza** [comment]
This is a simple example of the app's build.gradle

```

   android {
       compileSdkVersion 29
   
       defaultConfig {
           applicationId "com.example.gordontest"
           ...
       }
   
       signingConfigs {
           release {
               storeFile file("keystores/release.keystore")
               storePassword "xxx"
               keyAlias "xxx"
               keyPassword "xxx"
           }
       }
   
       buildTypes {
           release {
               minifyEnabled true
               signingConfig signingConfigs.release
               proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
           }
       }
   }
   ...

```

of course keystore passwords and so will not be hardcoded there, but for the sake of simplicity they are in this example.

**Comment by joshschriever** [comment]
Hmm, I still haven't been able to reproduce this problem, even with a simple example project with signingConfigs. @evaristokbza Maybe you could push your whole example project to a public repo so I can try to reproduce it with that?

**Comment by evaristokbza** [comment]
I investigated a bit more and realized my example was not correct. Sorry about that. The error actually happens when a signing config is applied to debug build type. Either by creating a signingConfig named "debug" or by explicitly doing it under buildTypes.

Following example should fail:

```
   android {
       compileSdkVersion 29
   
       defaultConfig {
           applicationId "com.example.gordontest"
           ...
       }
   
       signingConfigs {
           release {
               storeFile file("keystores/release.keystore")
               storePassword "xxx"
               keyAlias "xxx"
               keyPassword "xxx"
           }
       }
   
       buildTypes {
           debug {
               signingConfig signingConfigs.release
           }
           release {
               minifyEnabled true
               signingConfig signingConfigs.release
               proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
           }
       }
   }
```

On the bright side, this means removing the signingConfig from debug build works around the problem.

**Comment by joshschriever** [comment]
Fixed in [v1.4.3](https://github.com/Banno/Gordon/releases/tag/1.4.3)
